spark2/src/main/java/org/apache/iceberg/spark/source/IcebergSource.java

```diff
@@ -137,9 +138,11 @@ protected Table findTable(DataSourceOptions options, Configuration conf) {
       HadoopTables tables = new HadoopTables(conf);
       return tables.load(path.get());
     } else {
-      HiveCatalog hiveCatalog = HiveCatalogs.loadCatalog(conf);
+      Catalog catalog = options.get("impl")
+          .map(impl -> CatalogUtil.loadCustomCatalog("custom", impl, options.asMap(), conf))
+          .orElseGet(() -> HiveCatalogs.loadCatalog(conf));
       TableIdentifier tableIdentifier = TableIdentifier.parse(path.get());
-      return hiveCatalog.loadTable(tableIdentifier);
+      return catalog.loadTable(tableIdentifier);
```

Comment:
I think we shouldn't change the behavior of `IcebergSource` in this PR. We want to change how this source works and route queries through a catalog, but I'm not sure that using `impl` is the right way to do it. Let's stick with `HiveCatalogs` for now and revisit this in a follow up.